Quake: init 0.3.0

Eliot00 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

legendofmiracles commented on pkgs/applications/misc/quake/default.nix:
```diff
@@ -0,0 +1,26 @@
+  buildInputs = [ openssl ];
+
+  meta = with lib; {
+    description = "A knowledge management meta-framework for geeks.";
```
```suggestion
    description = "A knowledge management meta-framework for geeks";
```
Descriptions never have full-stops at the end

legendofmiracles reviewed (commented):
Your PR shouldn't target `nixos:release-21.11` but rather `master`. If you want it in 21.11 we can backport it

veprbl commented on pkgs/top-level/all-packages.nix:
```diff
@@ -1863,6 +1863,8 @@ with pkgs;
 
   passExtensions = recurseIntoAttrs pass.extensions;
 
+  quake = callPackage ../applications/misc/quake { };
```
```suggestion
  phodal-quake = callPackage ../applications/misc/phodal-quake { };
```

Eliot00 replied:
The quake repo will move to this [public org](https://github.com/inherd), can I just use the name `quake`?

veprbl replied:
It's not forbidden, but can cause a confusion https://repology.org/project/quake/versions

Eliot00 replied:
Thanks for reply, for avoid ambiguity, I will use org's name as prefix, `inherd-quake`

Eliot00 committed:
maintainers: add elliot

Eliot00 committed:
quake: init at 0.3.0

Update pkgs/applications/misc/inherd-quake/default.nix

Co-authored-by: Bobby Rong <rjl931189261@126.com>

Update pkgs/applications/misc/inherd-quake/default.nix

Co-authored-by: Bobby Rong <rjl931189261@126.com>

Update pkgs/applications/misc/inherd-quake/default.nix

Co-authored-by: Bobby Rong <rjl931189261@126.com>

Update pkgs/applications/misc/inherd-quake/default.nix

Co-authored-by: Bobby Rong <rjl931189261@126.com>

bobby285271 commented on pkgs/applications/misc/inherd-quake/default.nix:
```diff
@@ -0,0 +1,38 @@
+    owner = "phodal";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256:1f7k68g18g3dpnrsmhgmz753bly1i3f4lmsljiyp9ap0c6w8ahgg";
```
Nit *(This is just my personal preference)*

```suggestion
    sha256 = "1f7k68g18g3dpnrsmhgmz753bly1i3f4lmsljiyp9ap0c6w8ahgg";
```

bobby285271 commented on pkgs/applications/misc/inherd-quake/default.nix:
```diff
@@ -0,0 +1,38 @@
+    sha256 = "sha256:1f7k68g18g3dpnrsmhgmz753bly1i3f4lmsljiyp9ap0c6w8ahgg";
+  };
+
+  cargoSha256 = "sha256:1yqj9rq770j116138bqn4ycggy13vvym1cz50myfddb9rjjzafrl";
```
```suggestion
  cargoSha256 = "1yqj9rq770j116138bqn4ycggy13vvym1cz50myfddb9rjjzafrl";
```

bobby285271 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -1915,6 +1915,8 @@ with pkgs;
 
   passExtensions = recurseIntoAttrs pass.extensions;
 
+  inherd-quake = callPackage ../applications/misc/inherd-quake { };
```
We commonly put this in top-level.

```suggestion
  inherd-quake = callPackage ../applications/misc/inherd-quake {
    inherit (darwin.apple_sdk.frameworks) CoreServices Security;
  };
```

bobby285271 commented on pkgs/applications/misc/inherd-quake/default.nix:
```diff
@@ -0,0 +1,38 @@
+, pkg-config
+, openssl
+, stdenv
+, darwin
```
```suggestion
, CoreServices
, Security
```

bobby285271 commented on pkgs/applications/misc/inherd-quake/default.nix:
```diff
@@ -0,0 +1,38 @@
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ] ++ (
+    lib.optional stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.Security
+      darwin.apple_sdk.frameworks.CoreServices
+    ]
+  );
```
Maybe use <code>lib.optional<b>s</b></code>?

```suggestion
  buildInputs = [
    openssl
  ] ++ lib.optionals stdenv.isDarwin [
    CoreServices
    Security
  ];
```

bobby285271 commented:
Seems that ofborg is on holiday at the moment...

@ofborg build inherd-quake

bobby285271 commented:
@ofborg build inherd-quake

bobby285271 reviewed (approved):
<img src="https://user-images.githubusercontent.com/20080233/147575027-6ab4eb38-cbb4-46e4-9979-4298aac51268.png" width=70%>

Diff looks good to me, builds and prints help menu.
